verilog/analysis/checkers/disable_statement_rule.cc

```diff
@@ -0,0 +1,128 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "verilog/analysis/checkers/disable_statement_rule.h"
+
+#include <set>
+#include <string>
+
+#include "absl/strings/str_cat.h"
+#include "absl/strings/string_view.h"
+#include "common/analysis/citation.h"
+#include "common/analysis/lint_rule_status.h"
+#include "common/analysis/matcher/bound_symbol_manager.h"
+#include "common/analysis/matcher/core_matchers.h"
+#include "common/analysis/matcher/matcher.h"
+#include "common/text/symbol.h"
+#include "common/text/syntax_tree_context.h"
+#include "verilog/CST/identifier.h"
+#include "verilog/CST/verilog_matchers.h"  // IWYU pragma: keep
+#include "verilog/CST/verilog_nonterminals.h"
+#include "verilog/analysis/descriptions.h"
+#include "verilog/analysis/lint_rule_registry.h"
+
+namespace verilog {
+namespace analysis {
+
+using verible::GetStyleGuideCitation;
+using verible::LintRuleStatus;
+using verible::LintViolation;
+using verible::SyntaxTreeContext;
+using verible::matcher::Matcher;
+
+VERILOG_REGISTER_LINT_RULE(DisableStatementNoLabelsRule);
+
+absl::string_view DisableStatementNoLabelsRule::Name() {
+  return "disable-statement";
+}
+const char DisableStatementNoLabelsRule::kTopic[] =
+    "disable-invalid-in-non-sequential";
+const char DisableStatementNoLabelsRule::kMessage[] =
+    "Invalid usage of disable statement. Preferred construction is: disable "
+    "fork;";
+const char DisableStatementNoLabelsRule::kMessageSeqBlock[] =
+    "Invalid usage of disable statement. Preferred construction is: disable "
+    "label_of_seq_block;";
+
+std::string DisableStatementNoLabelsRule::GetDescription(
+    DescriptionType description_type) {
+  return absl::StrCat(
+      "Checks that there are no occurrences of ",
+      Codify("disable some_label", description_type),
+      " if label is referring to a fork or other none sequential block label.",
+      ". Use ", Codify("disable fork", description_type), " instead. See ",
+      GetStyleGuideCitation(kTopic), ".");
+}
+
+static const Matcher& DisableMatcher() {
+  static const Matcher matcher(
+      NodekDisableStatement(DisableStatementHasLabel()));
+  return matcher;
+}
+
+void DisableStatementNoLabelsRule::HandleSymbol(
+    const verible::Symbol& symbol, const SyntaxTreeContext& context) {
+  verible::matcher::BoundSymbolManager manager;
+  if (DisableMatcher().Matches(symbol, &manager)) {
```

Comment:
If this `if` is going over the whole method, consider an early-out here

```c++
 if (!DisableMatcher().Matches(symbol, &manager))
     return;
```

That way, the reader does not have to keep the surrounding branch on their 'reading stack', thus reducing complexity. Also, you have one more indentation level to work with :)